Document workaround for Firefox third-party cookie blocking (#109)
stackblitz/docs: docs/platform/webcontainers/browser-config.md

https://[*.]webcontainer.io
```

> For instance, in Chrome, go to `chrome://settings/cookies` and add those exceptions in the “Sites that can always use cookies” section.

Author: Both Chrome and Firefox do not allow navigating from a web page to a URL using their internal scheme (`chrome://` in Chrome, `about:` in Firefox).

I think this used to work, which is why I made this a link for users's convenience, but it's probably been blocked for security reasons.

:::

## Firefox: enabling Service Workers {#firefox-service-workers}

When Firefox’s Enhanced Tracking Protection is configured in “Custom” mode with the cookie blocking option set to “All cross-site cookies”, Firefox will block the Service Workers used by WebContainers.

You can work around this issue in a couple ways:

> 1. Use the “Strict” mode instead of “Custom” mode. It’s slightly more permissive than “Custom” mode with very strict options, so if you’d rather not, you can…

Reviewer: ```suggestion
1. Use the “Strict” mode instead of “Custom” mode. It’s important to note that it's slightly more permissive than “Custom” mode with very strict options. If this is not something you'd like to allow, you can...
```

> 2. Add exceptions to cookie blocking for a couple domains used to run StackBlitz projects.

Reviewer: ```suggestion
2. Add exceptions to cookie blocking for the domains used to run StackBlitz projects.
```

> First, visit `about:preferences#privacy` to check what your current settings look like. 

Reviewer: ```suggestion
To do so, visit `about:preferences#privacy` to check what your current settings look like. 
```

<img
  alt="The Firefox “Privacy & Security” settings page showing that Enhanced Tracking Protection is set to “Custom”, and to “Cookies: All cross-site cookies (may cause websites to break)”."
  src="./assets/firefox-settings-cookies-1.png"
  width="1024"
/>

To add exceptions for StackBlitz, scroll down to the “Cookies and Site Data” section, click on “Manage Exceptions…”, and add exceptions for the following sites:

```
https://stackblitz.io
https://webcontainer.io
```

<img
  alt="The Cookies and site data exceptions management window in Firefox settings. It shows a couple entries for stackblitz.io and webcontainer.io, both with the value “Allow”."
  src="./assets/firefox-settings-cookies-2.png"
  width="1024"
/>

> Don’t forget to click “Save Changes”. Then you should be able to reload the tab with your StackBlitz project, and hopefully everything should work now!

Reviewer: ```suggestion
Don’t forget to click “Save Changes”. Then you should be able to reload the tab with your StackBlitz project, and hopefully everything should work!
```

To allow StackBlitz projects to run, either set this option to “Balanced” or add exceptions for a couple domains used by StackBlitz projects.

> 1. Go to `edge://settings/privacy` → “Enhance your security on the web”.

Author: I haven't tested again in Edge, but given that it's based on Chromium I expect that the same linking restrictions apply.
